### DISCUSSION TITLE
How to filter detection result for specific locations?

### DISCUSSION THREAD

**Discussion by ly1616** [ultralytics/ultralytics#7411] [Q&A]
hello friends, I got something difficult to me and need help (⊙︿⊙)
basically I can get the ordinary result now, but in my scene I only need the result appears in the right edge of the picture, e.g. as the following attaches. All the results in the other positions are invalid actually.

how can I tell the model to filter and show the rest valid result or mark them automatically or any other way?

![image](https://github.com/ultralytics/ultralytics/assets/30145572/c66020b8-2fb4-48bb-ae15-0a032a3048cc)
![image](https://github.com/ultralytics/ultralytics/assets/30145572/64fb96ec-f601-49f7-92af-c018dc900798)

PS:  no matter and not sure how big the area of target is, 10% or 90% of the pic is ok too, the only request is "very close to the right edge of the pic"  

Thanks a lot ♪(･ω･)ﾉ

**Comment by ly1616** [comment]
I see a way but a bit verbose:
set 'save_txt=true' and read coordinates numbers in the result txts then select the qualified targets.
feels perfect~?

**Reply by IgorGoldberg** [reply]
You do not have to process txt files to filter detections by their locations
see this example how you can iterate over results and process them
https://github.com/ultralytics/ultralytics/issues/5180#issuecomment-1741971795
and also my question here:
https://github.com/ultralytics/ultralytics/issues/7348

Long story short in the "model" variable you can find the boxes coordinates

**Reply by pderrenger** [reply]
@IgorGoldberg hello! 😊 It looks like you're on the right track with processing the detection results. Instead of saving to `.txt` and reading from there, you can directly filter the results in Python after prediction. Here's a concise way to do it:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('path/to/your/model.pt')

# Perform prediction
results = model.predict(source='path/to/your/image.jpg')

# Filter detections based on x-coordinate (right edge of the image)
width = results.imgs[0].shape[1]  # Get the width of the image
edge_threshold = 0.9  # Set the threshold (e.g., 90% of the width)
filtered_boxes = []

# Iterate over detections and filter
for box in results.boxes.xyxy[0]:  # Assuming single image prediction
    x1, y1, x2, y2, conf, cls = box
    if x2 > edge_threshold * width:  # Check if box is close to the right edge
        filtered_boxes.append(box)

# Now 'filtered_boxes' contains only the detections near the right edge
```

This way, you can directly work with the detection results without the need for intermediate `.txt` files. Adjust the `edge_threshold` as needed for your specific use case. Happy coding! 🚀
